Euphorolog.Services/Services/AuthService.cs

```diff
@@ -20,44 +27,93 @@ public class AuthService : IAuthService
         public readonly EuphorologContext _context;
         public readonly IUsersRepository _usersRepository;
         public readonly IConfiguration _configuration;
-        public AuthService(EuphorologContext context, IUsersRepository usersRepository,IConfiguration configuration)
+        public readonly IHttpContextAccessor _httpContextAccessor;
+        public readonly IMapper _mapper;
+        public readonly MainDTOValidator<LogInRequestDTO> _logInValidator;
+        public readonly MainDTOValidator<SignUpRequestDTO> _signUpValidator;
+        public AuthService(
+            EuphorologContext context,
+            IUsersRepository usersRepository,
+            IConfiguration configuration,
+            IHttpContextAccessor httpContextAccessor,
+            IMapper mapper,
+            MainDTOValidator<LogInRequestDTO> logInValidator,
+            MainDTOValidator<SignUpRequestDTO> signUpValidator
+        )
         {
             _context = context;
             _usersRepository = usersRepository;
             _configuration = configuration;
+            _httpContextAccessor = httpContextAccessor;
+            _mapper = mapper;
+            _logInValidator = logInValidator;
+            _signUpValidator = signUpValidator;
         }
 
-        public async Task<Users> SignUp(Users user)
+        public async Task<SignUpResponseDTO> SignUp(SignUpRequestDTO req)
         {
-            if (await _usersRepository.UserExists(user.userName))
+            _signUpValidator.ValidateDTO(req);
+            if (await _usersRepository.UserExists(req.userName))
             {
-                throw new Exception("User already exists!");
+                throw new BadRequestException("User already exists!");
             }
-            var _utils = new Utilities(_configuration);
+            if (await _usersRepository.EmailUsed(req.eMail))
+            {
+                throw new BadRequestException("Email already used!");
+            }
+            var user = _mapper.Map<Users>(req);
+            var _utils = new Utilities(_configuration, _httpContextAccessor);
             _utils.CreatePasswordHash(user.password, out byte[] passwordHash, out byte[] passwordSalt);
             user.passwordHash = passwordHash;
             user.passwordSalt = passwordSalt;
             user.passChangedAt = DateTime.UtcNow;
-            user.passChangedflag = true;
-            return await _usersRepository.SignUp(user);
+            var userInfo = await _usersRepository.SignUp(user);
+            var ret = _mapper.Map<SignUpResponseDTO>(userInfo);
+            ret.token = _utils.CreateJWTToken(userInfo);
```

Comment:
So we are doing automatic login during Signup? Interesting approach. Some websites do that.